src/az_ulib_ipc/az_ulib_ipc.c

```diff
@@ -37,39 +37,87 @@ typedef union
  */
 static az_ulib_ipc* volatile _az_ipc_cb = NULL;
 
-static _az_ulib_ipc_interface* get_interface(
-    az_span name,
-    az_ulib_version version,
-    az_ulib_version_match_criteria match_criteria)
+/*
+ * This function follow the rules define in az_ulib_ipc_try_get_interface().
+ */
+static az_result get_interface(
+    az_span package_name,
+    az_ulib_version package_version,
+    az_span interface_name,
+    az_ulib_version interface_version,
+    _az_ulib_ipc_interface** handle)
 {
-  _az_ulib_ipc_interface* result = NULL;
+  _az_ulib_ipc_interface* interface_handle = NULL;
+  az_result result = AZ_ERROR_ITEM_NOT_FOUND;
```

Comment:
>az_result result = AZ_ERROR_ITEM_NOT_FOUND; [](http://example.com/codeflow?start=2&length=43)

can you move this to 
```
else
{
result = AZ_ERROR_ITEM_NOT_FOUND;
}
```

Reply:
I fighted with this pre-initialization, but I didn't find any other solution. There are issues like:
1. The compiler will complain about the test in the line 116 because the for loop in 54 can have 0 iterations.
2. I cannot use the interface_handle == NULL to define the result because in the Ambiguous case, the interface_handle will not be NULL.
3. Use the else that you propose for the if in the line 60 will only work if the required interface is the last one list.

So, I decided to make an exception here.

Reply:
sounds good!